tests/test_tokenization_fast.py

```diff
@@ -258,281 +238,147 @@ def assert_build_inputs_with_special_tokens(self, tokenizer_r, tokenizer_p):
-
-            # Check for padding
-            self.assert_padding(tokenizer_r, tokenizer_p)
-
-    def test_gpt2(self):
-        for tokenizer_name in GPT2Tokenizer.pretrained_vocab_files_map["vocab_file"].keys():
-            tokenizer_p = GPT2Tokenizer.from_pretrained(tokenizer_name)
-            tokenizer_r = GPT2TokenizerFast.from_pretrained(tokenizer_name)
-
-            # Check we have the same number of added_tokens for both pair and non-pair inputs.
-            self.assertEqual(tokenizer_r.num_added_tokens(False), tokenizer_p.num_added_tokens(False))
-            self.assertEqual(tokenizer_r.num_added_tokens(True), tokenizer_p.num_added_tokens(True))
-
-            # Check we have the correct max_length for both pair and non-pair inputs.
-            self.assertEqual(tokenizer_r.max_len_single_sentence, tokenizer_p.max_len_single_sentence)
-            self.assertEqual(tokenizer_r.max_len_sentences_pair, tokenizer_p.max_len_sentences_pair)
-
-            # Assert the set of special tokens match.
-            self.assertSequenceEqual(
-                tokenizer_p.special_tokens_map.items(),
-                tokenizer_r.special_tokens_map.items(),
-                "GPT2 tokenizers doesn't have the same set of special_tokens",
+            self.assertEqual(
+                tokenizer_p.batch_encode_plus([" "], add_special_tokens=should_add_special_tokens),
+                tokenizer_r.batch_encode_plus([" "], add_special_tokens=should_add_special_tokens),
             )
 
-            # Assure tokenization overlap between python and rust impl.
-            self.assert_tokenization_python_rust_almost_equals(tokenizer_p, tokenizer_r, 0.0)
-
-            # Ensure add_tokens and add_special_tokens return the correct vocab size
-            self.assert_add_tokens(tokenizer_r)
-
-            # Check for offsets mapping
-            self.assert_offsets_mapping(tokenizer_r)
-
-            # Check for dynamic encoding sequence handling in batch_encode_plus
-            self.assertRaises(ValueError, self.assert_batch_encode_dynamic_overflowing, tokenizer_r)
+    def assert_padding(self, tokenizer_r, tokenizer_p, max_length=15):
+        def assert_padded_input_match(input_r: list, input_p: list, max_length: int):
 
-            # Check alignment for build_inputs_with_special_tokens
-            self.assert_build_inputs_with_special_tokens(tokenizer_r, tokenizer_p)
+            # Ensure we match max_length
+            self.assertEqual(len(input_r), max_length), self.assertEqual(len(input_p), max_length)
 
-            # Check the number of returned files for save_vocabulary
-            self.assert_save_pretrained(tokenizer_r, tokenizer_p)
+            # Ensure the number of padded tokens is the same
+            padded_tokens_r = list(takewhile(lambda i: i == tokenizer_r.pad_token_id, reversed(input_r)))
+            padded_tokens_p = list(takewhile(lambda i: i == tokenizer_p.pad_token_id, reversed(input_p)))
```

Comment:
We usually try to use lambdas as little as possible, as they're usually a bit hard to read. cc @thomwolf

Reply:
hmm here I think it's fine, no?

Reply:
In tests I'm more fine with them

Reply:
👍